consensus/src/error.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum ConsensusError {
```

Comment:
Suggest adding fields to the error variants to give more insightful information

Reply:
can you be specific on which variant? as some of the errors returned are cases where we get a None value which i map to a unit variant